wicket-core/src/main/java/org/apache/wicket/ajax/AbstractDefaultAjaxBehavior.java

```diff
@@ -81,7 +82,27 @@
 	@Override
 	protected void onBind()
 	{
-		getComponent().setOutputMarkupId(true);
+		final Component component = getComponent();
+		
+		component.setOutputMarkupId(true);
+		
+		if (getStatelessHint(component))
+		{
+			//generate behavior id
+	        component.getBehaviorId(this);
+		}
+	}
+	
+	@Override
+	public CharSequence getCallbackUrl()
+	{
+		if (getStatelessHint(getComponent()))
+		{
+		    final Url url = Url.parse(super.getCallbackUrl().toString());
+	        return UrlUtils.mergeParameters(url).toString();
```

Comment:
#mergeParameters() does nothing when no parameters are passed - are the page parameters missing here?